File: benchmark/nodejs/cumulative_sum.js
const n = 100000;
const arr = [];
for (let i = 1; i <= n; i++) {
    arr.push(i);
}
const start = Date.now();
const cum = [];
let s = 0;
for (let i = 0; i < n; i++) {
    s = s + arr[i];
    cum.push(s);
}
const end = Date.now();
console.log(`Last: ${cum[n - 1]}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/cumulative_sum.rb
n = 100000
arr = []
(1..n).each { |i| arr << i }
start = Time.now.to_f * 1000
cum = []
s = 0
n.times do |i|
    s = s + arr[i]
    cum << s
end
finish = Time.now.to_f * 1000
puts "Last: #{cum[n - 1]}"
puts "Time: #{(finish - start).round}ms"

